systems/framework/lcm_driven_loop.h

```diff
@@ -206,7 +227,13 @@ class LcmDrivenLoop {
 
         simulator_->AdvanceTo(time);
         // Force-publish via the diagram
-        diagram_ptr_->Publish(diagram_context);
+        /// Notice that diagram.Publish() is for dispatching the publish of
+        /// TriggerType::kForced type. In LcmPublisherSystem, both periodic and
+        /// per-step publishes are also forced publish.
+        /// https://github.com/RobotLocomotion/drake/blob/03fe7e4/systems/lcm/lcm_publisher_system.h#L54-L56
```

Comment:
This is not an entirely correct interpretation of LcmPublisherSystem--I fixed this issue in Drake some time ago. See the correct constructor https://github.com/RobotLocomotion/drake/blob/master/systems/lcm/lcm_publisher_system.h#L111 which allows you to specify exactly which publish types to use (e.g. forced only)

Reply:
It's a little bit ambiguous here. What I meant was we cannot run both `AdvanceTo()` and `Publish()` in the case of periodic or per-step publisher. 

After reading the code for the second time, I should probably add a flag which allows users to run `Publish()`